src/crypto/crypto.gni

```diff
@@ -16,4 +16,7 @@ declare_args() {
   # Crypto implementation: mbedtls, openssl, tinycrypt, boringssl, platform.
   chip_crypto = ""
   chip_with_se05x = 0
+
+  # Compile mbedtls externally. Only used if chip_crypto == "mbedtls"
+  chip_external_mbedtls = false
```

Comment:
how about using the `current_os == "zephyr"` condition here? Otherwise it seems zephyr will not be able to override while all others would be.

Reply:
Seems like EFR32 fails because `current_os` is undefined. The builder script uses `--args` which I think clears all default build args. EFR32 platform then imports `crypto.gni` without having target_os/current_os.

Changed it so instead the three configs that use zephyr os will directly set `chip_external_mbedtls = true`. Lmk if this is preferred. It should simplify things for users of this flag and users of `--args`.

Reply:
can we assert that `!chip_external_mbedtls || (chip_crypto == "mbedtls")` ?

Reply:
current_os should not be undefined, can you please check this?

Reply:
Added the assert.

For some reason it's undefined, here is a failing check: https://github.com/alexhqwang/connectedhomeip/actions/runs/3414393855/jobs/5682287901
I was also able to reproduce it locally. When I add prints for `current_os` or even `target_os` in various EFR32 build files, it shows the same error. Should EFR32 be fixed to define these instead? I thought it would be better to change on the zephyr side, since other downstreams could have the same issue.